ComponentKit/LayoutComponents/CKStackUnpositionedLayout.mm

```diff
@@ -15,10 +15,38 @@
 #import <ComponentKit/CKAssert.h>
 
 #import "ComponentUtilities.h"
+#import "CKComponentInternal.h"
 #import "CKComponentSubclass.h"
 #import "CKStackLayoutComponentUtilities.h"
 
 /**
+ 
+ */
+static CGFloat resolveCrossDimensionMaxForStretchChild(const CKStackLayoutComponentStyle &style,
+                                                       const CKStackLayoutComponentChild &child,
+                                                       const CGFloat stackMax,
+                                                       const CGFloat crossMax)
+{
+  // stretched children may have a cross direction max that is smaller than the minimum size constraint of the parent.
+  const CGFloat computedMax = (style.direction == CKStackLayoutDirectionVertical ?
+                               child.component.size.resolve(kCKComponentParentSizeUndefined).max.width :
+                               child.component.size.resolve(kCKComponentParentSizeUndefined).max.height);
+  return computedMax == INFINITY ? crossMax : computedMax;
```

Comment:
Interesting. So if the computed maximum is infinity then we should stretch child? Otherwise use the computed maximum?